expr/helper/helper.go

```diff
@@ -147,11 +147,11 @@ func AggregateSeries(name string, args []*types.MetricData, absent_if_first_seri
 }
 
 // SummarizeValues summarizes values
-func SummarizeValues(f string, values []float64) (float64, bool) {
+func SummarizeValues(f string, values []float64) (error, float64, bool) {
 	rv := 0.0
 
 	if len(values) == 0 {
-		return 0, true
+		return nil, 0, true
 	}
 
 	switch f {
```

Comment:
Looks like, we have a difference vs the python version https://graphite.readthedocs.io/en/latest/functions.html?highlight=summarize#graphite.render.functions.summarize I don't see `total` being a summarization type, and `average` is used instead of `avg`.

Reply:
I think the documentation is a bit off.  I was reading the source code, and this is what graphite is using:

https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/functions/aggfuncs.py

`total` is an alias for `sum`.
Added `average` as an alias for `avg`.

Another difference that I noticed is that draphite doesn't seem to support percentiles(see the default: in our case).

Reply:
Also added `current` as alias for `last`.
Filled https://github.com/bookingcom/carbonapi/issues/323 for the missing aggregations.

Reply:
@avereha Does "current" mean the same to you as "last"? I'd avoid using "current" if it's not in the docs. It's misleading.

Reply:
As I mentioned before, documentation for `summarize` is not up to date.
The functions supported by `summarize` are exactly the same function supported by `aggregate`.
This list of functions is more up to date: https://graphite.readthedocs.io/en/latest/functions.html?highlight=summarize#graphite.render.functions.aggregate
And yes, `current` == `last`.